jax/_src/lax/parallel.py

```diff
@@ -1122,9 +1122,23 @@ def _pbroadcast_lowering(ctx, x, *, axis_name, source):
   def source_to_front(group):
     return [group[source]] + list(group[:source]) + list(group[source + 1:])
   replica_groups = [source_to_front(group) for group in replica_groups]
+  is_spmd = isinstance(
+      ctx.module_context.axis_context,
+      (SPMDAxisContext, ShardingContext),
+  )
+  if is_spmd:
+    # We want to emit the collective-broadcast with global device IDs and a unique
+    # channel ID, as otherwise it interprets the devices as replicas instead
+    # of partitions - and XLA is configured with only a single replica.
+    channel = ctx.module_context.new_channel()
+    channel_handle = hlo.ChannelHandle.get(channel, mlir.DEVICE_TO_DEVICE_TYPE)
+    other_args = dict(channel_handle=channel_handle)
+  else:
+    other_args = {}
   channel = ctx.module_context.new_channel()
```

Comment:
You don't need this channel anymore since it's defined on line 1133. Can you please remove it? Then I can get this in.

Reply:
Extra line removed.